appengine_config.py

```diff
@@ -58,3 +62,52 @@
 # six python path to the app engine vendor first.
 import six # isort:skip  pylint: disable=wrong-import-position, wrong-import-order
 reload(six) # pylint: disable=reload-builtin
+
+
+# For some reason, pkg_resources.get_distribution returns an empty list in the
+# prod env. We need to monkeypatch this function so that prod deployments
+# work. Otherwise, pkg_resources.get_distribution('google-api-core').version in
+# google/api_core/__init__.py throws a DistributionNotFound error, and
+# similarly for pkg_resources.get_distribution("google-cloud-tasks").version in
+# google/cloud/tasks_v2/gapic/cloud_tasks_client.py, which results in the
+# entire application being broken on production.
+import requests_toolbelt.adapters.appengine # isort:skip  pylint: disable=wrong-import-position, wrong-import-order
+requests_toolbelt.adapters.appengine.monkeypatch()
+old_get_distribution = pkg_resources.get_distribution # pylint: disable=invalid-name
+
+
+class MockDistribution(python_utils.OBJECT):
+    """Mock distribution object for the monkeypatching function."""
+
+    def __init__(self, version):
+        self.version = version
+
+
+def monkeypatched_get_distribution(distribution_tuple):
+    """Monkeypatched version of pkg_resources.get_distribution.
```

Comment:
Link to the issue comment that proposed this hack.

Reply:
Done.